pkg/kubelet/prober/worker.go

```diff
@@ -131,11 +134,21 @@ func newWorker(
 func (w *worker) run() {
 	probeTickerPeriod := time.Duration(w.spec.PeriodSeconds) * time.Second
 
+	// Ensure jitterSeed is between 1 and 2
+	jitterSeed := 1 + w.randomFloat - math.Floor(w.randomFloat)
+
 	// If kubelet restarted the probes could be started in rapid succession.
 	// Let the worker wait for a random portion of tickerPeriod before probing.
 	// Do it only if the kubelet has started recently.
 	if probeTickerPeriod > time.Since(w.probeManager.start) {
-		time.Sleep(time.Duration(rand.Float64() * float64(probeTickerPeriod)))
+		jitter := time.Duration(jitterSeed*float64(w.spec.PeriodSeconds)) * time.Second
+		// Override: readiness worker should start not later than startup worker,
+		// otherwise it will miss the manual trigger of the probe once started.
+		if w.probeType == readiness && w.container.StartupProbe != nil {
+			jitter = time.Duration(jitterSeed*
+				math.Min(float64(w.spec.PeriodSeconds), float64(w.container.StartupProbe.PeriodSeconds))) * time.Second
+		}
+		time.Sleep(jitter)
```

Comment:
I'd suggest to sleep here for `jitter` + `InitialDelaySeconds`. This will spread out startup of workers even better, plus it will ensure that for the small initialDelay and large period, we will not wait for the entire period and will start probes faster.

Reply:
so you suggest to sleep for min(all defined *Probe.PeriodSeconds) * randomFloat + InitialDelaySeconds
but which InitialDelaySeconds should I use?
the goal is to make sure "readiness worker should start not later than startup worker"